beacon-chain/sync/regular_sync_test.go

```diff
@@ -741,3 +743,98 @@ func TestHandleStateReq_OK(t *testing.T) {
 	}
 	testutil.AssertLogsContain(t, hook, "Sending finalized, justified, and canonical states to peer")
 }
+
+func TestCanonicalBlockList_CanRetrieveCanonical(t *testing.T) {
+	db := internal.SetupDB(t)
+	defer internal.TeardownDB(t, db)
+	ss := setupService(db)
+
+	// Construct the following chain:
+	//    /- B3
+	//	 B1  - B2 - B4
```

Comment:
I think this is messed up?

Reply:
It looks good in IDE, might be github messing it up